### PR TITLE
Export all control plane metrics under a common prefix

### PR DESCRIPTION
**Pull request by mceldeen**
Common prefix: `kubernetes.controlplane`

### CODE REVIEW

**Comment by mmichael** [comment]
Closing PR. After discussing with product manager, we decided that we'd need to follow the deprecation process to change prefixes of existing metrics. We'll reconsider these changes at a later date

File: deploy/kubernetes/4-collector-config.yaml [hunk 1]
```diff
@@ -303,47 +307,41 @@ data:
...
-        scheme: http
-        prefix: kube.dns.
-        filters:
-          metricAllowList:
-          - 'kube.dns.http.request.duration.microseconds'
-          - 'kube.dns.http.request.size.bytes'
-          - 'kube.dns.http.requests.total.counter'
-          - 'kube.dns.http.response.size.bytes'
-          - 'kube.dns.kubedns.dnsmasq.*'
-          - 'kube.dns.process.*'
```
Comment (priyaselvaganesan) [inline deploy/kubernetes/4-collector-config.yaml:325]: Would like clarification for the sake of the newly opened PR #366: 
What is the reason for commenting the dns and coredns metrics here? Need to know if the same needs to be done in the new PR #366 as well.

File: hack/test/dashboards/deploy/control-plane-collector-config.yaml [hunk 2]
```diff
@@ -303,28 +303,6 @@ data:
       # Kubernetes component auto discovery
       ##########################################################################
```
Comment (priyaselvaganesan) [inline hack/test/dashboards/deploy/control-plane-collector-config.yaml:305]: Need clarification here before porting to PR #366: Why are we removing the `dns` and `coredns` metrics  below? Is that a necessity for deploying to demo environment in vmware sandbox?

File: hack/test/overlays/test-kind/metrics/additional.jsonl [hunk 3]
```diff
@@ -1 +1,3 @@
+{"Name":"kubernetes.controlplane.coredns.dns.request.duration.seconds.bucket","Tags":{"source":"","cluster":"","le":"","namespace":"","pod":"","server":"","type":"","zone":""}}
```
Comment (priyaselvaganesan) [inline hack/test/overlays/test-kind/metrics/additional.jsonl:1]: Need clarification before porting this change PR #366: Does having coredns metrics in the hack/test/files/metrics.jsonl break the test in gke and eks? Is that why it is put explicitly in kind's additional metrics?

**Pull request closed without merging**